sync/sync_head.go

```diff
@@ -114,13 +114,8 @@ func (s *Syncer[H]) subjectiveHead(ctx context.Context) (H, error) {
 
 // setSubjectiveHead takes already validated head and sets it as the new sync target.
 func (s *Syncer[H]) setSubjectiveHead(ctx context.Context, netHead H) {
-	// TODO(@Wondertan): Right now, we can only store adjacent headers, instead we should:
-	//  * Allow storing any valid header here in Store
-	//  * Remove ErrNonAdjacent
-	//  * Remove writeHead from the canonical store implementation
```

Comment:
Looks like `writeHead` can stay, the only things that needs to be changed is to be sure that it's the highest header.